### PR TITLE
OSSM-5393 Add product-specific configuration

### PR DESCRIPTION
**Pull request by dgn**
This creates a new ossm/ directory that we can use to store everything related to only downstream. It also adds a Makefile.vendor.mk to overwrite some variables in the Makefile. All other changes have been generated using `make gen`.

### CODE REVIEW

File: Makefile.vendor.mk [hunk 1]
```diff
@@ -0,0 +1,6 @@
+VERSION = 3.0.0
+OPERATOR_NAME = servicemeshoperator3
+HUB = quay.io/maistra-dev
+CHANNELS = 3.0
```
Comment (FilipB) [inline Makefile.vendor.mk:4]: Do we want to use 'candidates' channel for now?

File: Makefile.vendor.mk [hunk 2]
```diff
@@ -0,0 +1,6 @@
+VERSION = 3.0.0
```
Comment (FilipB) [inline Makefile.vendor.mk:1]: Do we want to use '3.0.0-tp.1' here or overwrite this in cpaas midstream?

File: Makefile.vendor.mk [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+OPERATOR_NAME = servicemeshoperator3
+HUB = quay.io/maistra-dev
+CHANNELS = 3.0
+HELM_VALUES_FILE = ossm/values.yaml
+VERSIONS_YAML_FILE = versions.yaml
```
Comment (FilipB) [inline Makefile.vendor.mk:6]: Do we want to remove versions which we don't build downstream here or in cpaas midstream?

**Review by FilipB — CHANGES_REQUESTED** [review]

File: Makefile.vendor.mk (hunk 2 above)
Comment (dgn) [reply Makefile.vendor.mk:1]: same here, I'd say we do that when we branch off into a release branch for tp1

File: Makefile.vendor.mk (hunk 3 above)
Comment (dgn) [reply Makefile.vendor.mk:6]: same here, I'd say we do that when we branch off into a release branch for tp1

**Review by FilipB — APPROVED** [review]

**Pull request merged**